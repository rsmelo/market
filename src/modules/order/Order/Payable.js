import React from 'react'
import PropTypes from 'prop-types'

const Payable = ({
  status,
  amount,
  fee,
  recipientId,
  paymentDate,
  dateCreated,
}) => (
  <div>
    <ul>
      <li>
        id do recebedor: ${recipientId}
      </li>
      <li>
        Valor: R$: ${amount}
      </li>
      <li>
        Taxas: R$: ${fee}
      </li>
      <li>
        Status: ${status}
      </li>
      <li>
        Data da criação : ${dateCreated}
      </li>
      <li>
        Data do pagamento : ${paymentDate}
      </li>
    </ul>
  </div>
)

Payable.propTypes = {
  status: PropTypes.string.isRequired,
  amount: PropTypes.number.isRequired,
  fee: PropTypes.number.isRequired,
  recipientId: PropTypes.string.isRequired,
  paymentDate: PropTypes.string.isRequired,
  dateCreated: PropTypes.string.isRequired,
}
export default Payable
