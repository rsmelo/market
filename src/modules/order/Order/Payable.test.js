import React from 'react'
import renderer from 'react-test-renderer'

import Payable from './Payable'

jest.mock('react-router-dom', () => ({ Link: 'Link' }))

describe('Order', () => {
  it('renders correctly', () => {
    const props = {
      status: 'waiting_funds',
      amount: 24000,
      fee: 6000,
      recipientId: 're_cj2wd5u2600fecw6eytgcbkd0',
      paymentDate: '2017-08-01T03:00:00.000',
      dateCreated: '2017-06-29T18:16:49.411Z',
    }
    const tree = renderer
      .create(<Payable {...props} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
