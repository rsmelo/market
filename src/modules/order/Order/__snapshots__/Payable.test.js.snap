// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order renders correctly 1`] = `
<div>
  <ul>
    <li>
      id do recebedor: $
      re_cj2wd5u2600fecw6eytgcbkd0
    </li>
    <li>
      Valor: R$: $
      24000
    </li>
    <li>
      Taxas: R$: $
      6000
    </li>
    <li>
      Status: $
      waiting_funds
    </li>
    <li>
      Data da criação : $
      2017-06-29T18:16:49.411Z
    </li>
    <li>
      Data do pagamento : $
      2017-08-01T03:00:00.000
    </li>
  </ul>
</div>
`;
